app/src/main/java/com/github/sdp/tarjetakuna/ui/scanner/ScannerFragment.kt

```diff
@@ -20,41 +36,276 @@ class ScannerFragment : Fragment() {
+            view?.let {
+                Snackbar.make(
+                    it,
+                    "Error: cannot take 2 pictures at the same time",
+                    Snackbar.LENGTH_LONG
+                ).show()
+            }
+            return
+        } else {
+            _takingPicture = true
+            if (!takePicture()) {
+                view?.let { Snackbar.make(it, "Error taking picture", Snackbar.LENGTH_LONG).show() }
+                _takingPicture = false
+            }
+        }
+    }
+
+    private fun takePicture(): Boolean {
+        // Get a stable reference of the modifiable image capture use case
+        val imageCapture = imageCapture ?: return false
+        val context = this.context ?: return false
+        val contentResolver = this.activity?.contentResolver ?: return false
+
+        // Create time stamped name and MediaStore entry.
+        val name = SimpleDateFormat(FILENAME_FORMAT, Locale.US)
+            .format(System.currentTimeMillis())
+        val contentValues = ContentValues().apply {
+            put(MediaStore.MediaColumns.DISPLAY_NAME, name)
+            put(MediaStore.MediaColumns.MIME_TYPE, "image/jpeg")
+            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P) {
+                put(MediaStore.Images.Media.RELATIVE_PATH, "Pictures/Tarjetakuna-Image")
+            }
+        }
+
+        // Create output options object which contains file + metadata
+        val outputOptions = ImageCapture.OutputFileOptions.Builder(
+            contentResolver,
+            MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+            contentValues
+        ).build()
+
+        // Set up image capture listener, which is triggered after photo has been taken
+        imageCapture.takePicture(
+            outputOptions,
+            ContextCompat.getMainExecutor(context),
+            object : ImageCapture.OnImageSavedCallback {
+                override fun onError(exc: ImageCaptureException) {
+                    view?.let {
+                        Snackbar.make(it, R.string.scanner_photo_failed, Snackbar.LENGTH_LONG)
+                            .show()
+                    }
+                    Log.e(TAG, getString(R.string.scanner_photo_failed), exc)
+                    _takingPicture = false
+                }
+
+                override fun onImageSaved(output: ImageCapture.OutputFileResults) {
+                    view?.let {
+                        Snackbar.make(it, R.string.scanner_photo_saved, Snackbar.LENGTH_LONG).show()
+                    }
+                    Log.d(TAG, getString(R.string.scanner_photo_saved) + output.savedUri)
+                    _takingPicture = false
+                }
+            }
+        )
+        return true
+    }
+
+    /**
+     * Start the camera and bind the "preview" = live preview on screen, "takePicture" = save the
+     * picture on the device, and "analysis" = text recognition and object detection to be done correctly
```

Comment:
why is saving the picture on the device and analyzing it done in the fragment?

Reply:
analysis is done at run time on the camera feed directly, we can save the picture to store it with the user card later on